### DISCUSSION TITLE
How to add a picture to the ATO report from CodeceptJS (from IDE)?

### DISCUSSION THREAD

**Discussion by blamaniq** [allure-framework/allure2#1826] [Questions & Support] [answered]
How to add a picture to the ATO report from manual test-cases in CodeceptJS (from IDE)? 
Is it possible to add via addAttachment? Show an example?
Purpose: Create manual test cases from IDE, There is a need to add pictures to AllureTestOps.

I'm looking for a solution like here:
https://codecept.io/plugins/#public-api
addAttachment(name, buffer, type) - add an attachment to current test / suite

**Comment by cheshi-mantu** [comment, marked as answer]
@blamaniq 
Hi, Maxim
Now, the goal setting is a bit unclear, so we need to word the user story: I'm assuming  you are trying to implement the concept of manual test cases as a code, and you want the attachments from a test result to land in a test case generated after the manual test case passed (launch created, test passed manually, the result is set, launch closed), is that correct?

**Reply by blamaniq** [reply]
Thanks!

**Reply by blamaniq** [reply]
I want to add an image inside attachment while a process of creation of manual tests as a code.

**Reply by blamaniq** [reply]
For now, it is possible to add an image inside an attachment using an utility fs (like that:
const fs = require('fs');

Scenario
xxxx

    await allure.createStep('xxx', async () => {
      const ltd4 = fs.readFileSync('manual/img.png');
      allure.createAttachment('ltd4', ltd4, 'image/png');
})

And the image is displayed inside AllureTO, but inside the Launch only (a test run result). If to close the Launch, and open the test case (that was created from the launch) - there is no image displayed. 

I want the image be displayed inside the test case, that was created via IDE (from code).

**Reply by cheshi-mantu** [reply]
@blamaniq
Hi Maxim,
an attachment is an attribute of test result, it is not transferred to a test case (test documentation) in Allure TestOps. So the behaviour you are facing is correct.
Should you need different behaviour, please do not hesitate to file a feature request via https://help.qameta.io using your **work email** address, please describe the business process requiring the transfer of a test result's attachment to a test case.

best

Egor
